File: FangZhiHu/app/src/main/java/network/download/DownInfo.java
package network.download;


import network.DownProgressListener;

/**
 * Created by yuantongqin on 2017/8/8.
 */

public class DownInfo {

    public long read;//下载的长度
    public long count;//总长度
    public boolean complete;//是个完成
    public String mDownloadUrl;//下载的url
    public DownState state;
    private String savaPath;

    private DownProgressListener mListener;
    public DownInfo(String url, String savePath){
        this.mDownloadUrl = url;
        this.savaPath = savePath;
    }


    public DownProgressListener getmListener() {
        return mListener;
    }

    public void setmListener(DownProgressListener mListener) {
        this.mListener = mListener;
    }

    public void setSavaPath(String savaPath) {
        this.savaPath = savaPath;
    }

    public String getSavaPath() {
        return savaPath;
    }

    public enum DownState {
        START,
        PAUSE,
        ERROR,
        COMPLETE,
    }
}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/download/DownInfo.kt
package network.download

/**
 * Created by tongqinyuan on 2018/3/7.
 */
data class DownInfo(var savaPath:String,var mDownloadUrl:String) {

    var read: Long = 0L//下载的长度
    var count: Long = 0L//总长度
    var complete: Boolean = false//是个完成
    var state:DownState? = null

    private var mListener :DownProgressListener<*>? = null

    fun getListener() :DownProgressListener<*>?{
        return mListener
    }

    fun setListener(listener: DownProgressListener<*>){
        this.mListener = listener
    }

    enum class DownState{
        START,
        PAUSE,
        ERROR,
        COMPLETE
    }


}